src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageDetailsTabViewModel.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public class PackageDetailsTabViewModel : ViewModelBase, IDisposable
+    {
+        private bool _readmeTabEnabled;
+        public PackageDetailsTabViewModel()
+        {
+            _readmeTabEnabled = true;
+            Tabs = new ObservableCollection<TabViewModelBase>();
+        }
+
+        public void Initialize(DetailControlModel detailControlModel, INuGetPackageFileService nugetPackageFileService, ItemFilter currentFilter, bool readmeTabAvailable)
+        {
+            ReadmePreviewViewModel = new ReadmePreviewViewModel(nugetPackageFileService, currentFilter);
+            ReadmePreviewViewModel.IsVisible = readmeTabAvailable;
+            DetailControlModel = detailControlModel;
+            _readmeTabEnabled = readmeTabAvailable;
+
+            Tabs.Add(ReadmePreviewViewModel);
+            Tabs.Add(DetailControlModel);
+
+            DetailControlModel.PropertyChanged += DetailControlModel_PropertyChanged;
+            ReadmePreviewViewModel.PropertyChanged += ReadMePreviewViewModel_PropertyChanged;
+        }
+
+        public async Task SetCurrentFilterAsync(ItemFilter filter)
+        {
+            if (_readmeTabEnabled)
+            {
+                await ReadmePreviewViewModel.SetCurrentFilterAsync(filter);
+            }
+        }
+
+        private void ReadMePreviewViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (ReadmePreviewViewModel is not null
+                && e.PropertyName == nameof(ReadmePreviewViewModel.ReadmeMarkdown))
+            {
+                Tabs[0].IsVisible = !string.IsNullOrWhiteSpace(ReadmePreviewViewModel.ReadmeMarkdown);
+                if (!Tabs[0].IsVisible)
+                {
+                    SelectedTab = Tabs[1];
+                }
```

Comment:
To make this code more resilient to changes in the future where we may want to add a new tab and not have to modify this code, could we just say that we want the selected tab to be whatever the first visible tab is?

I also now wonder if this is going to conflict with the user settings. It would be nice to have the tab selection logic all in one place.

Disclaimer: I did not test this suggestion and wrote it quickly in the browser. Please verify correctness.

```suggestion
                foreach (var tab in Tabs) 
                {
                    if (tab.IsVisible)
                    {
                        SelectedTab = tab;
                        break;
                    }
                }
```